parser.y

```diff
@@ -5858,6 +5860,32 @@ FunctionCallNonKeyword:
 			Args:   []ast.ExprNode{$6, $4, direction},
 		}
 	}
+|	weightString '(' Expression ')'
+	{
+		$$ = &ast.FuncCallExpr{
+			FnName: model.NewCIStr($1),
+			Args:   []ast.ExprNode{$3},
+		}
+	}
+|	weightString '(' Expression "AS" Char FieldLen ')'
+	{
+		$$ = &ast.FuncCallExpr{
+			FnName: model.NewCIStr($1),
+			Args:   []ast.ExprNode{$3, ast.NewValueExpr("CHAR"), ast.NewValueExpr($6)},
+		}
+	}
+|	weightString '(' Expression "AS" "BINARY" FieldLen ')'
+	{
+		$$ = &ast.FuncCallExpr{
+			FnName: model.NewCIStr($1),
+			Args:   []ast.ExprNode{$3, ast.NewValueExpr("BINARY"), ast.NewValueExpr($6)},
+		}
+	}
+/** The following syntax is defined in MySQL parser, but not described in MySQL reference manual, we do not support it for now.
```

Comment:
I don't think we need to support it, since we don't support `flag`.
It's strange to support a grammer not described in MySQL reference manual but don't support a grammer  described in MySQL reference manual.

Reply:
So what's you suggesion, remove this?

Reply:
Yeah,

Reply:
Done.